dotnetv3/cross-service/PhotoAssetManager/PamApiAnnotations/DetectLabelsFunction.cs

```
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+using Amazon;
+using Amazon.DynamoDBv2;
+using Amazon.DynamoDBv2.DataModel;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.S3Events;
+using Amazon.Rekognition;
+using Amazon.S3;
+using Amazon.Util;
+using PamServices;
+
+// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
+//[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+
+namespace PamApiAnnotations;
+
+/// <summary>
+/// The function code for detecting and storing labels for an uploaded image.
+/// </summary>
+public class DetectLabelsFunction
+{
+    IAmazonS3 S3Client { get; set; }
+    private readonly ImageService _imageService;
+    private readonly LabelService _labelService;
+
+    /// <summary>
+    /// Default constructor. This constructor is used by Lambda to construct the instance. When invoked in a Lambda environment
+    /// the AWS credentials will come from the IAM role associated with the function and the AWS region will be set to the
+    /// region the Lambda function is executed in.
+    /// </summary>
+    public DetectLabelsFunction()
+    {
+        S3Client = new AmazonS3Client();
+        _imageService = new ImageService(new AmazonRekognitionClient());
+        _labelService = new LabelService(new DynamoDBContext(new AmazonDynamoDBClient()));
+
+        var labelsTableName = Environment.GetEnvironmentVariable("LABELS_TABLE_NAME");
+
+        if (labelsTableName != null)
+        {
+            AWSConfigsDynamoDB.Context.AddMapping(new TypeMapping(typeof(Label), labelsTableName));
+        }
```

Same concern as above with silent failures.

---

Resolved with comment above.